### ISSUE TITLE
Add a Route::fallbackView() method

### ISSUE THREAD

**Issue by carestad** [laravel/ideas#1961]
## Problem 

```php
// routes/web.php:
Route::fallback(function() {
  return view('my-404');
}
```

... isn't cacheable with `artisan routes:cache`.

```php
// routes/web.php:
Route::fallback('MyController@404');

// app/Http/Controllers/MyController.php
public function 404() {
    return view('my-404');
}
```

... is cacheable, but I end up having this needless boilerplate code in my controller.

## Suggestion 

Add a `Route::fallbackView(string $view, array $data = []) {}` method, similar to `Route::view()` today, which is cacheable with `artisan routes:cache`.

**Comment by carestad** [comment]
I am however, seeing a need for being able to set the status code. I use fallback for my 404 pages, and would like to set 404 as a status code.

**Comment by carestad** [comment]
I have a [branche where I have implemented this](https://github.com/laravel/framework/compare/6.x...carestad:feature/route-fallback-view), but I am not sure about the changed in `ViewController`. It returns a response now instead of an actual view like before.

**Comment by martinbean** [comment]
The status code shouldn’t be configurable. You’re either defining a fallback route when no other matches (404) or you’re not.

**Comment by carestad** [comment]
> The status code shouldn’t be configurable. You’re either defining a fallback route when no other matches (404) or you’re not.

Yeah, that's sort of what I was wondering. Today, the only route definition where you can also set the status code for, are redirect routes.

But at the same time, fallback routes are thought to be used for 404s right? So leaving them with a 200 status would be a bit funky as well? That isn't a problem for the normal `fallback()` as you can just set the status in the closure or controller method
